crates/sqlexec/src/planner/logical_plan/client_exchange.rs

```diff
@@ -0,0 +1,162 @@
+use super::*;
+use uuid::Uuid;
+
+// TODO: Probably should be treated as solely physical. Currently some things
+// are left unimplemented since this node won't be sent through rpc.
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub struct ClientExchangeSend {
+    pub broadcast_id: Uuid,
+    pub input: DfLogicalPlan,
+}
+
+impl TryFrom<protogen::sqlexec::logical_plan::ClientExchangeSend> for ClientExchangeSend {
+    type Error = ProtoConvError;
+
+    fn try_from(
+        _proto: protogen::sqlexec::logical_plan::ClientExchangeSend,
+    ) -> Result<Self, Self::Error> {
+        unimplemented!("unknown semantics for deserializing client send")
+    }
+}
+
+impl UserDefinedLogicalNodeCore for ClientExchangeSend {
+    fn name(&self) -> &str {
+        Self::EXTENSION_NAME
+    }
+
+    fn inputs(&self) -> Vec<&DfLogicalPlan> {
+        vec![&self.input]
+    }
+
+    fn schema(&self) -> &datafusion::common::DFSchemaRef {
+        unimplemented!("unknown semantics for deserializing client send")
+    }
+
+    fn expressions(&self) -> Vec<datafusion::prelude::Expr> {
+        vec![]
+    }
+
+    fn fmt_for_explain(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}", Self::EXTENSION_NAME)
+    }
+
+    fn from_template(
+        &self,
+        _exprs: &[datafusion::prelude::Expr],
+        _inputs: &[DfLogicalPlan],
+    ) -> Self {
+        self.clone()
+    }
+}
+
+impl ExtensionNode for ClientExchangeSend {
+    const EXTENSION_NAME: &'static str = "ClientExchangeSend";
+
+    fn try_decode_extension(extension: &LogicalPlanExtension) -> Result<Self> {
+        match extension.node.as_any().downcast_ref::<Self>() {
+            Some(s) => Ok(s.clone()),
+            None => Err(internal!(
+                "ClientExchange::try_from_extension: unsupported extension",
+            )),
+        }
+    }
+
+    fn try_encode(&self, buf: &mut Vec<u8>, _codec: &dyn LogicalExtensionCodec) -> Result<()> {
+        use protogen::sqlexec::logical_plan as protogen;
+
+        let client_exchange = protogen::ClientExchangeSend {
+            broadcast_id: self.broadcast_id.into_bytes().to_vec(),
+        };
+
+        let extension = protogen::LogicalPlanExtensionType::ClientExchangeSend(client_exchange);
+
+        let lp_extension = protogen::LogicalPlanExtension {
+            inner: Some(extension),
+        };
+
+        lp_extension
+            .encode(buf)
+            .map_err(|e| internal!("{}", e.to_string()))?;
+
+        Ok(())
+    }
+}
```

Comment:
so i know we don't have the physical planning fully in place yet, but i don't think we'd want a `ClientExchangeSend` node at the logical level. Just checking to see if this is just part of the draft, or do you see a case for having some of this information at the logical level?

Reply:
Yep I agree. Fully plan on these getting removed when the physical plan stuff is in place.